packages/vite/src/node/env.ts

```diff
@@ -1,9 +1,13 @@
 import fs from 'node:fs'
-import dotenv from 'dotenv'
-import dotenvExpand from 'dotenv-expand'
+import { config, parse } from 'dotenv'
+import { expand } from 'dotenv-expand'
 import { arraify, lookupFile } from './utils'
 import type { UserConfig } from './config'
 
+config({
+  debug: process.env.DEBUG?.includes('vite:dotenv')
+})
```

Comment:
Doesn't this load the env file into the process Vite is running in? I think we shouldn't do that by default.

It doesn't look like `parse` support the `debug` option now but we could skip it.